src/app/app-routing.module.ts

```diff
@@ -2,14 +2,20 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 
+//404 import
+import {NotFoundComponent} from './not-found/not-found.component';
+
 /**
  * Fallback to this route when no prior route is matched.
  */
 const routes: Routes = [
+  {
+    path: '404',
+    component: NotFoundComponent
+  },
   {
     path: '**',
-    redirectTo: '',
-    pathMatch: 'full'
+    redirectTo: '/404'
```

Comment:
Instead of a redirect and a new path for 404 update the code with this:
```typescript
{
    path: '**',
    component: PageNotFoundComponent
}
```